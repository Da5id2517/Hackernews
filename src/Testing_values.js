export const list = [
  {
    title: 'Naslov',
    url: 'https://lemonparty.org',
    author: 'SomeBloke',
    num_comment: 3,
    points: 4,
    objectID: 0
  },
  {
    title: 'Seheron',
    url: 'https://youtube.org',
    author: 'Tseries',
    num_comment: 12,
    points: 3,
    objectID: 1
  }
]

export const blank_func = () => {};
export const test_string = 'test_string';
