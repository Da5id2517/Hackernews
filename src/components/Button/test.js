import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Button from './';
import {blank_func, test_string} from '../../Testing_values.js';


describe('Button', () => {
  it('renders properly.', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <Button onClick={blank_func}>
      {test_string}
      </Button>,
       div
     );
    ReactDOM.unmountComponentAtNode(div);
  });

  it('has valid snap.', () => {
    const comp = renderer.create(
      <Button onClick={blank_func}>
      {test_string}
      </Button>
    );
    let tree = comp.toJSON();
    expect(tree).toMatchSnapshot();
  })
})
