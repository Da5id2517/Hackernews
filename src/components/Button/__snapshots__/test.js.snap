// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button has valid snap. 1`] = `
<button
  className=""
  onClick={[Function]}
  type="button"
>
  test_string
</button>
`;
